### PR TITLE
Role options integration for Server Access

### PR DESCRIPTION
**Pull request by xacrimon**
This PR lays out some groundwork and defines options required for general use and SSH access as per https://github.com/gravitational/predicate-lang/issues/43

### CODE REVIEW

**Comment by xacrimon** [comment]
@klizhentas ptal

**Review by flyinghermit — APPROVED** [review]
LGTM, but get another approval before merging.

**Comment by xacrimon** [comment]
@klizhentas friendly ping

**Review by klizhentas — APPROVED** [review]

**Pull request merged**
